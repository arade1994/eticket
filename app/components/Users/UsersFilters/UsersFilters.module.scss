.usersFilters {
  display: flex;
  flex-direction: row;
  gap: 1rem;
  align-items: center;
  flex-wrap: wrap;

  & .searchInput {
    position: relative;
    width: 300px;
    flex: 0 0 auto;

    & .searchIcon {
      position: absolute;
      left: 1rem;
      top: 50%;
      transform: translateY(-50%);
      color: #666;
      font-size: 1rem;
      pointer-events: none;
    }

    & .searchInputField {
      width: 100%;
      padding: 0.75rem 1rem 0.75rem 2.75rem;
      border: 1px solid #e0e0e0;
      border-radius: 8px;
      font-size: 0.95rem;
      background: #f8f9fa;
      transition: all 0.3s ease;
      box-sizing: border-box;

      &:focus {
        outline: none;
        border-color: #006064;
        box-shadow: 0 0 0 2px rgba(0, 96, 100, 0.1);
      }

      &::placeholder {
        color: #999;
      }
    }
  }

  & .resetFiltersBtn {
    display: flex;
    align-items: center;
    gap: 0.5rem;
    padding: 0.75rem 1rem;
    border: none;
    border-radius: 8px;
    background: #fff5f5;
    color: #dc3545;
    font-size: 0.95rem;
    font-weight: 500;
    cursor: pointer;
    transition: all 0.3s ease;
    flex: 0 0 auto;

    & .resetIcon {
      font-size: 0.9rem;
    }

    &:hover {
      background: #ffe5e5;
      transform: translateY(-1px);
    }

    &:active {
      transform: translateY(0);
    }
  }
}
