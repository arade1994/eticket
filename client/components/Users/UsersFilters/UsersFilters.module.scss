.usersFilters {
  display: flex;
  flex-direction: row;
  gap: 15px;

  & .searchInput {
    height: 30px;
    font-size: 14px;
  }

  & .resetFiltersBtn {
    cursor: pointer;
    align-self: center;
    color: rgb(172, 14, 14);

    &:hover {
      color: rgb(107, 11, 11);
    }
  }
}
