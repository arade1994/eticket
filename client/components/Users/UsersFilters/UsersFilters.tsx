import { ChangeEvent } from "react";
import classes from "./UsersFilters.module.scss";

interface Props {
  searchText: string;
  onChangeSearchText: (event: ChangeEvent<HTMLInputElement>) => void;
  onResetFilters: () => void;
}

const UsersFilters: React.FC<React.PropsWithChildren<Props>> = ({
  searchText,
  onChangeSearchText,
  onResetFilters,
}) => {
  const isFiltersApplied = searchText.trim() !== "";

  return (
    <div className={classes.usersFilters}>
      <input
        id="search"
        className={classes.searchInput}
        type="text"
        placeholder="Insert first name or last"
        value={searchText}
        onChange={onChangeSearchText}
      />
      {isFiltersApplied && (
        <div className={classes.resetFiltersBtn} onClick={onResetFilters}>
          Reset filters
        </div>
      )}
    </div>
  );
};

export default UsersFilters;
