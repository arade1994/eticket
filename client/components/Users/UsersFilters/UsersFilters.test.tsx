import { render, screen } from "@testing-library/react";
import { describe, expect, test, vi } from "vitest";
import UsersFilters from "./UsersFilters";
import userEvent from "@testing-library/user-event";

const mockOnChangeSearchText = vi.fn();
const mockOnResetFilters = vi.fn();

const renderUsersFilters = (searchText: string) =>
  render(
    <UsersFilters
      searchText={searchText}
      onChangeSearchText={mockOnChangeSearchText}
      onResetFilters={mockOnResetFilters}
    />
  );

describe("<UsersFilters />", () => {
  test("it should match a snapshot when rendered by default", () => {
    const { baseElement } = renderUsersFilters("");

    expect(baseElement).toMatchSnapshot();
  });

  test("it should render search filter", async () => {
    renderUsersFilters("");

    const searchInput = screen.getByTestId("search");
    expect(searchInput).toBeDefined();
    await userEvent.type(searchInput, "Test");
    expect(mockOnChangeSearchText).toHaveBeenCalled();
  });

  test("it should render reset filters button when filters are applied", async () => {
    renderUsersFilters("Test");

    const resetFiltersBtn = screen.getByText("Reset filters");
    await userEvent.click(resetFiltersBtn);
    expect(mockOnResetFilters).toHaveBeenCalled();
  });
});
