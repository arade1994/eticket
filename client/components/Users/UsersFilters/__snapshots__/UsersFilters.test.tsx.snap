// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<UsersFilters /> > it should match a snapshot when rendered by default 1`] = `
<body>
  <div>
    <div
      class="_usersFilters_9c49f3"
    >
      <input
        class="_searchInput_9c49f3"
        id="search"
        placeholder="Insert first name or last"
        type="text"
        value=""
      />
    </div>
  </div>
</body>
`;
